source/config/Config.cpp

```diff
@@ -277,13 +277,12 @@ bool PlainConfig::LoadFromCliArgs(const CliArgs &cliArgs)
         thingName = cliArgs.at(PlainConfig::CLI_THING_NAME).c_str();
     }
 
-    bool loadFeatureCliArgs = tunneling.LoadFromCliArgs(cliArgs) && logConfig.LoadFromCliArgs(cliArgs);
+    bool loadFeatureCliArgs = tunneling.LoadFromCliArgs(cliArgs) && logConfig.LoadFromCliArgs(cliArgs) && httpProxyConfig.LoadFromCliArgs(cliArgs);
```

Comment:
you will also have to validate the HTTP proxy config. Can we update it over here? https://github.com/awslabs/aws-iot-device-client/blob/main/source/config/Config.cpp#L2689-L2708